api/models/model.py

```diff
@@ -155,6 +156,28 @@ def getByDeploymentStatus(self, deployment_status):
     def getByPublishStatus(self, publish_status):
         return self.dbs.query(Model).filter(Model.is_published == publish_status).all()
 
+    def getByInTheLoopStatus(self, in_the_loop):
+        if in_the_loop:
+            return (
+                self.dbs.query(Model)
+                .join(Round, Round.tid == Model.tid)
+                .filter(Round.url.contains(Model.endpoint_name))
+                .distinct()
+                .all()
+            )
+        else:
+            sub_query = (
+                self.dbs.query(Model.id)
+                .join(Round, Round.tid == Model.tid)
+                .filter(Round.url.contains(Model.endpoint_name))
+                .distinct()
+                .subquery()
```

Comment:
This is a little complicated, but the only way I found I could detect models in the loop, is by looking at all rounds for a tasks, looking at the `url` field (which should be a bunch of model endpoint names separated by `|`)

Reply:
So the only way to detect models not in the loop is by:
(1) finding all the models in the loop
(2) choosing all the models that are not in this list

Reply:
Yeah unfortunately I don't think that there is a better way.